app/models/manageiq/providers/cisco_intersight/physical_infra_manager/operations/power.rb

```diff
@@ -36,62 +28,59 @@ def restart_to_sys_setup(_args, _options)
-    end
-
-    # Select any supported method of powering the server down.
-    def power_down(server, _options)
-      trigger_first_valid_power_action(server, %w[ForceOff GracefulShutdown])
-    end
-
-    # Select any supported method of powering the server up.
-    def power_up(server, _options)
-      trigger_first_valid_power_action(server, %w[On ForceOn])
+      reset_server(server, "Reboot");
     end
 
     private
 
-    def trigger_first_valid_power_action(server, rtypes)
-      server.with_provider_object do |system|
-        available_rtypes = get_available_rtypes(system)
-        rtype = rtypes.find { |t| available_rtypes.include?(t) }
-        if rtype.nil?
-          _log.error("#{rtypes} and #{available_rtypes} are disjunct.")
-          raise MiqException::Error, "No acceptable reset type"
+    def reset_server(server, power_state)
+      # Changing the admin_power_state attribute in IntersightClient::ComputeServerSetting object according to power_state
+      # Possible values for admin_power_state:
+      # @see https://github.com/xlab-si/intersight-sdk-ruby/blob/45ec426b9d061502166499a755ac119058002972/lib/intersight_client/models/compute_server_setting.rb#L28
+      #   - admin_power_state == `Policy` - Power state is set to the default value in the policy
+      #     [Meaning: look at result of IntersightClient::PowerApi.new.get_power_policy_list (select the relevant element according to moid of the server)]
+      #   - admin_power_state == `PowerOn` - Power state of the server is set to On.
+      #   - admin_power_state == `PowerOff` - Power state of the server is set to Off.
+      #   - admin_power_state == `PowerCycle` - Power state of the server is reset.
+      #   - admin_power_state == `HardReset` - Power state of the server is hard reset.
+      #   - admin_power_state == `Shutdown` - Operating system on the server is shut down.
+      #   - admin_power_state == `Reboot` - Power state of IMC is rebooted.
+
+      _log.info("Requesting #{power_state} for #{server.ems_ref}.")
+
+      with_provider_connection do |_client|
+        compute_api = IntersightClient::ComputeApi.new
+        system = compute_api.get_compute_physical_summary_by_moid(server.ems_ref)
+
+        # Get the related ComputeServerSettings:
+        # This only works for servers that are Intersight-managed and have a directly related IntersightClient::ComputeServerSetting object
+        compute_server_settings_list = compute_api.get_compute_server_setting_list({:filter => "(Server.Moid eq '#{server.ems_ref}')"}).results
+
+        if compute_server_settings_list.empty?
+          raise MiqException::Error, "No IntersightClient::ComputeServerSetting object found for ems_ref #{server.ems_ref} . Server might not be Intersight-managed."
+          return
+        end
+        if compute_server_settings_list.size > 1
+          raise MiqException::Error, "Multiple IntersightClient::ComputeServerSetting retrieved for ems_ref #{server.ems_ref}."
```

Comment:
Is this actually possible?  If so seems like we need to re-evaluate what ems_ref is

Reply:
This should not be the case, because we are supposed to be using Intersight specific uniqu identifiers.e